### PR TITLE
BFS article completed do check 

### PR DESCRIPTION
**Pull request by Ruchip16**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 1]
```diff
@@ -0,0 +1,341 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+## Introduction
+
+Breadth-first search (BFS) is an algorithm for traversing or searching tree or graph data structures.It is similar to level-order traversal in trees. It starts at the root node and explores all of the neighbor nodes at the present node before moving on to the nodes at the next depth level.BFS uses queue data structure for traversing the nodes.
```
Comment (nagasrihith608) [inline docs/graph-algorithms/breadth-first-search.mdx:7]: give space after the full stop in the first line.

write it as  "a queue" in the last line

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+
+## Explanation
+
+The BFS traversal of Graphs also traverses the graph in levels. It starts the traversal with a given `source vertex (V)`, visits all of the vertices adjacent to the initially given vertex and `pushes them all to a queue` in order of visiting and `pops an element from the front of the queue`, visits all of its neighbours and pushes the neighbours which are not already visited into the queue and repeats the process until the queue is empty or all of the vertices are visited.
```
Comment (nagasrihith608) [inline docs/graph-algorithms/breadth-first-search.mdx:12]: The theory seems fine but can you change the sentence formation as it sounds a bit vague.

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+**Step 5]**![graph-5.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs13.jpg)
+
+
+## Algorithm
+
+- Create a `boolean array` say `visited[]`.
+- Create a `queue`, mark the source vertex visited as `visited[s] = true` and push it into the queue.
+- Loop until the `queue is not empty`, repeat the below steps:
+   * Pop an element from the queue and print the popped element.
+   * Traverse all of the vertices adjacent to the vertex poped from the queue.
```
Comment (nagasrihith608) [inline docs/graph-algorithms/breadth-first-search.mdx:34]: correct popped spelling

Comment (nagasrihith608) [inline docs/graph-algorithms/breadth-first-search.mdx:34]: small correction in the line: vertexes are

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+**Step 4]**![graph-4.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs12.jpg )
+
+**Step 5]**![graph-5.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs13.jpg)
+
+
+## Algorithm
+
+- Create a `boolean array` say `visited[]`.
+- Create a `queue`, mark the source vertex visited as `visited[s] = true` and push it into the queue.
```
Comment (nagasrihith608) [inline docs/graph-algorithms/breadth-first-search.mdx:31]: add a comma after true

File: docs/graph-algorithms/breadth-first-search.mdx (hunk 1 above)
Comment (nagasrihith608) [reply docs/graph-algorithms/breadth-first-search.mdx:7]: you can write it as : 

Breadth-first search (BFS) is an algorithm to traverse/search the data structure tree or a graph.

**Review by nagasrihith608 — CHANGES_REQUESTED** [review]

**Comment by nagasrihith608** [comment]
Hi @Ruchip16, just a small suggestion. You could use Grammarly as it helps in finding all the grammatical and spelling mistakes :)

File: docs/graph-algorithms/breadth-first-search.mdx (hunk 1 above)
Comment (Ruchip16) [reply docs/graph-algorithms/breadth-first-search.mdx]: I have run the theory through Grammarly once and done the changes that you suggested and committed them, do check once

**Comment by Ruchip16** [comment]
@nagasrihith608 I have run the theory through Grammarly once and done the changes that you suggested and committed them, do check PR now

File: .theia/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "editor.autoSave": "on"
+}
```
Comment (HarshCasper) [inline .theia/settings.json:3]: You shouldn't have this here

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+
+**Step 1]**![graph-1.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs9.jpg )
+
+**Step 2]**![graph-2.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs10.jpg )
+
+**Step 3]**![graph-3.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs11.jpg )
+
+**Step 4]**![graph-4.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs12.jpg )
+
+**Step 5]**![graph-5.png]( https://www.guru99.com/images/1/020820_0543_BreadthFirs13.jpg)
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:25]: Don't use images that you didn't create.

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+## Introduction
+
+Breadth-first search (BFS) is an algorithm to traverse/search the data structure tree or a graph. It is similar to level-order traversal in trees. It starts at the root node and explores all of the neighbor nodes at the present node before moving on to the nodes at the next depth level. BFS uses a **'queue'** data structure for traversing the nodes.
+
+
+## Explanation
+
+The BFS traversal of the graph is similar to the level-order traversal in trees. It starts the traversal with a given `source vertex (V)`, visits all of the vertices adjacent to the given vertex and `pushes them all to a queue` in order of visiting and `pops it from the front of the queue`, visits all adjacent nodes and pushes them into the queue. This process is repeated until the queue is empty or all of the vertices are visited.
+
+### Let's have one example:
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:14]: Keep examples within the explanation section

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+
+
+## Algorithm
+
+- Create a `boolean array` say `visited[]`.
+- Create a `queue`, mark the source vertex visited as `visited[s] = true`, and push it into the queue.
+- Loop until the `queue is not empty`, repeat the below steps:
+   * Pop an element from the queue and print the popped element.
+   * Traverse all of the vertices adjacent to the vertex popped from the queue.
+   * If any of the adjacent vertexes are not already visited, mark it visited and push it to the queue.
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:35]: Keep a line break

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+            x = path.get(x, -1)
+
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+### Input:
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:309]: Keep Heading-4

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+Enter the number of edges in the Graph : 
+4
+Enter the edges :
+0 1
+1 2
+0 2
+2 3
+```
+
+### Output:
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:323]: Keep Heading-4

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Ruchip16** [comment]
@HarshCasper @nagasrihith608 I will do the necessary changes within tomorrow

File: .theia/settings.json (hunk 5 above)
Comment (Ruchip16) [reply .theia/settings.json:3]: Means I should comment it out ?? I didn't understand actually like what are you trying to say

**Comment by Ruchip16** [comment]
I have made the changes required hope that helps now

File: .theia/settings.json [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// {
+//     "editor.autoSave": "on"
+// }
```
Comment (HarshCasper) [inline .theia/settings.json:3]: Just remove this file here

File: .vscode/settings.json [hunk 12]
```diff
@@ -0,0 +1,3 @@
+{
+    "editor.autoSave": "on"
+}
```
Comment (HarshCasper) [inline .vscode/settings.json:3]: Just remove this file here

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 13]
```diff
@@ -0,0 +1,346 @@
...
+
+**Step 1]**![BFS 1](https://user-images.githubusercontent.com/72685035/115226603-0ede9300-a12d-11eb-81f6-7947fa9a3366.jpg) 
+
+**Step 2]**![BFS 2](https://user-images.githubusercontent.com/72685035/115226742-36cdf680-a12d-11eb-83c9-018e1654a0fb.jpg) 
+
+**Step 3]**![BFS 3](https://user-images.githubusercontent.com/72685035/115226830-4cdbb700-a12d-11eb-8fe2-37987f6d1ff1.jpg) 
+
+**Step 4]**![BFS 4](https://user-images.githubusercontent.com/72685035/115226837-5107d480-a12d-11eb-89ca-8ce70a1c3583.jpg)
+
+**Step 5]**![BFS 5](https://user-images.githubusercontent.com/72685035/115226850-549b5b80-a12d-11eb-9c48-bccb2a9a831c.jpg)
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:24]: Did you make these diagrams?

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 14]
```diff
@@ -0,0 +1,346 @@
...
+slug: /graph-algorithms/breadth-first-search
+---
+## Introduction
+
+Breadth-first search (BFS) is an algorithm to traverse/search the data structure tree or a graph. It is similar to level-order traversal in trees. It starts at the root node and explores all of the neighbor nodes at the present node before moving on to the nodes at the next depth level. BFS uses a **'queue'** data structure for traversing the nodes.
+
+
+## Explanation
+
+- The BFS traversal of the graph is similar to the level-order traversal in trees. It starts the traversal with a given `source vertex (V)`, visits all of the vertices adjacent to the given vertex and `pushes them all to a queue` in order of visiting and `pops it from the front of the queue`, visits all adjacent nodes and pushes them into the queue. This process is repeated until the queue is empty or all of the vertices are visited.
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:12]: Give a line break after this

File: docs/graph-algorithms/breadth-first-search.mdx [hunk 15]
```diff
@@ -0,0 +1,346 @@
...
+
+    def bfs(self, sv, ev):
+        visited = [False for j in range(self.vertices)]
+        path = dict()
+        if self.bfsHelper(sv, ev, visited, path) is True:
+            return path
+        return dict()
+
+if __name__ == "__main__":
+    print("Enter no of vertices and edges-")
```
Comment (HarshCasper) [inline docs/graph-algorithms/breadth-first-search.mdx:292]: The sample I/O of the languages you have here doesn't seem to match at all

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: .theia/settings.json (hunk 11 above)
Comment (Ruchip16) [reply .theia/settings.json:3]: okay

File: docs/graph-algorithms/breadth-first-search.mdx (hunk 13 above)
Comment (Ruchip16) [reply docs/graph-algorithms/breadth-first-search.mdx:24]: No, but they are not copied either, so they can be included

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity since last 21 days. Thank you for your valuable time.😊

**Pull request closed without merging**